src/OrchardCore.Modules/OrchardCore.Contents/Scripting/ContentMethodsProvider.cs

```diff
@@ -50,10 +53,12 @@ public ContentMethodsProvider()
                 Method = serviceProvider => (Action<ContentItem, object>)((contentItem, properties) =>
                 {
                     var contentManager = serviceProvider.GetRequiredService<IContentManager>();
-                    var props = JObject.FromObject(properties);
-                    var content = (JObject)contentItem.ContentItem.Content;
 
-                    content.Merge(props, new JsonMergeSettings { MergeArrayHandling = MergeArrayHandling.Replace });
+                    var json = JsonConvert.SerializeObject(properties);
+                    contentItem.Apply(JsonConvert.DeserializeObject<ContentItem>(json));
+                    if (!string.IsNullOrWhiteSpace(contentItem.Content?.TitlePart?.Title?.ToString()))
+                        contentItem.DisplayText = contentItem.Content.TitlePart.Title?.ToString();
+
```

Comment:
This is good, but we need to allow for the `ContentItem.DisplayText` to be set in the json as well as the TitlePart.

We also need to cover of the other well known properties.

See https://github.com/OrchardCMS/OrchardCore/issues/4964 and https://github.com/OrchardCMS/OrchardCore/issues/4963

Suggest we should have a helper that can test if the `properties` have a well known property name and if so apply it, as well as the `contentItem.Apply` to effect elements (such as the `TitlePart`).

I wonder if the `DisplayText` shouldn't be updated in the `TitlePart` handler, rather than as it is in the `DisplayDriver`

Reply:
I don't trust any change that touches "one" property like this. Also TitlePart doesn't store the display text, it should not store anything at all since we introduced this `DisplayText` property.

Reply:
I'm surprised we require changes in those method as I tested them last week and everything seemed to work fine.

Reply:
Here is the data I was trying to update through that update method through a workflow.

```
{
  "ContentItemId": "4yaf4fhsn34ka0nzqewec26sme",
  "ContentItemVersionId": "46j5pyt07zw6dsm7b8y277hknc",
  "ContentType": "Acme",
  "DisplayText": "Acme #1 edited",
  "Owner": "admin",
  "Author": "admin",
  "Acme": {},
  "AutoroutePart": {
    "Path": "acme-1",
    "SetHomepage": false
  },
  "Teams": {
    "ContentItems": [
      {
        "ContentItemId": "40jj6t8mx68883v1frbdyrcnwf",
        "ContentItemVersionId": null,
        "ContentType": "Team",
        "DisplayText": "Team #1 edited",
        "Latest": false,
        "Published": false,
        "ModifiedUtc": "2019-12-24T08:02:44.6490803Z",
        "PublishedUtc": null,
        "CreatedUtc": null,
        "Owner": null,
        "Author": "admin",
        "Team": {}
      },
      {
        "ContentItemId": "4m8kdrf2avv6bwyck5h7xmbada",
        "ContentItemVersionId": null,
        "ContentType": "Team",
        "DisplayText": "Team #2 edited",
        "Latest": false,
        "Published": false,
        "ModifiedUtc": "2019-12-24T08:02:44.6759843Z",
        "PublishedUtc": null,
        "CreatedUtc": null,
        "Owner": null,
        "Author": "admin",
        "Team": {}
      }
    ]
  }
}
```

Reply:
![screencapture-localhost-44300-Admin-Workflows-Types-Edit-23-2019-12-24-02_34_31](https://user-images.githubusercontent.com/3228637/71615863-63997b80-2b81-11ea-914e-a5d1ff32a957.png)

Reply:
The proper fix was to add .GetAwaiter().GetResult(); on the contentManager.UpdateAsync().
Though, if there is a different use case than what I showed above, let us know so that we can test it.
I have absolutely not tested everything, but to me it makes totally sense to retrieve the content item that you want to update as a workflow task step before updating it.

Reply:
that's what i wanted to say, so obvious but i did not see it. Missing awaiter. I will double check with my workflow and if that was the case i will close this PR